### ISSUE TITLE
[YSQL][LST] FATAL: Check failed: _s.ok() Bad status: Invalid argument (yb/tserver/pg_client_service.cc:377): Unknown session: 97

### ISSUE THREAD

**Issue by def-** [yugabyte/yugabyte-db#13511]
Jira Link: [DB-3106](https://yugabyte.atlassian.net/browse/DB-3106)
### Description

```
$ cd ~/code/yugabyte-db
$ git checkout a166e5a6e87d1a9b9e72406e26a8d927933e6ea0
$ arc patch D18465
$ ./yb_build.sh release
$ bin/yb-ctl --replication_factor 3 create --tserver_flags=yb_enable_read_committed_isolation=true,ysql_num_shards_per_tserver=1,enable_stream_compression=true,stream_compression_algo=1,yb_num_shards_per_tserver=1 --master_flags=yb_enable_read_committed_isolation=true,enable_stream_compression=true,stream_compression_algo=1,enable_automatic_tablet_splitting=true,tablet_split_low_phase_shard_count_per_node=40,tablet_split_high_phase_shard_count_per_node=50
$ cd ~/code/yb-long-system-test
$ git checkout 2027af8b
$ while true; do python3.9 ./long_system_test.py --nodes=127.0.0.1:5433,127.0.0.2:5433,127.0.0.3:5433 --threads=10 --runtime=60 --max-columns=20 --complexity=full --seed=370648; done
```
Ran into this fatal:
```
[deen@devp yugabyte-db]$ cat /home/deen/yugabyte-data/node-1/disk-1/yb-data/tserver/logs/yb-tserver.FATAL.details.2022-08-02T11_00_22.pid889849.txt
F20220802 11:00:22 ../../../../../../src/yb/yql/pggate/pggate.cc:1683] Check failed: _s.ok() Bad status: Invalid argument (yb/tserver/pg_client_service.cc:377): Unknown session: 97
    @     0x7fabfe0756ab  google::LogMessage::SendToLog()
    @     0x7fabfe0760ce  google::LogMessage::Flush()
    @     0x7fabfe07885f  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fac02879a8f  yb::pggate::PgApiImpl::ClearSeparateDdlTxnMode()
    @           0x9e31c0  YBTxnDdlProcessUtility
    @           0x856a8a  PortalRunUtility
    @           0x856177  PortalRunMulti
    @           0x855a36  PortalRun
    @           0x8531b6  yb_exec_simple_query_impl
    @           0x853756  yb_exec_query_wrapper_one_attempt
    @           0x8502cb  PostgresMain
    @           0x7bc3cc  BackendRun
    @           0x7bb820  ServerLoop
    @           0x7b7ddb  PostmasterMain
    @           0x7185ed  PostgresServerProcessMain
    @           0x718a92  main
    @     0x7fac02142825  __libc_start_main
    @           0x4c67a9  _start
```
This might be related to https://github.com/yugabyte/yugabyte-db/issues/13495 since afterwards I am seeing the same symptom as in #13495 when trying to drop database:
```
psycopg2.errors.ObjectInUse: database "db_lst_370648" is being accessed by other users
DETAIL:  There are 2 other sessions using the database.
```
Logs from LST:
```
2022-08-02 10:55:20,860 worker_6   ERROR    Unexpected query failure: InternalError_
Query: SELECT '(,82]'::INT4RANGE, -3.6625943250856636 FROM tg0_1 ORDER BY 2 DESC LIMIT 7;
  values: None
  runtime: 2022-08-02 10:55:15.913 - 2022-08-02 10:55:20.860
  supports explain: True
  supports rollback: True
  affected rows: None
Action: RandomSelectAction
Error class: InternalError_
Error code: XX000
Error message: ERROR:  Query error: GetTransactionStatus RPC (request call id 20319907) to 127.0.0.1:9100 timed out after 5.000s
    @           0x9b4038  errmsg
    @           0x9df825  HandleYBStatusAtErrorLevel
    @           0x56e68f  ybFetchNext
    @           0x6f63d8  ybcIterateForeignScan
    @           0x6eae5f  ForeignNext
    @           0x6c214d  ExecScan
    @           0x6db836  ExecLimit
    @           0x6b9473  standard_ExecutorRun
    @     0x7fabfd905ae5  pgss_ExecutorRun
    @     0x7fabfd8fc60a  ybpgm_ExecutorRun
    @           0x855e9b  PortalRunSelect
    @           0x8559da  PortalRun
    @           0x8531b6  yb_exec_simple_query_impl
    @           0x853756  yb_exec_query_wrapper_one_attempt
    @           0x8502cb  PostgresMain
    @           0x7bc3cc  BackendRun
    @           0x7bb820  ServerLoop
    @           0x7b7ddb  PostmasterMain
    @           0x7185ed  PostgresServerProcessMain
    @           0x718a92  main
    @     0x7fac02142825  __libc_start_main
    @           0x4c67a9  _start
Transaction isolation level: read uncommitted
DB Node: host: 127.0.0.1, port: 5433
DB Backend PID: 887790
2022-08-02 10:55:30,941 MainThread INFO
2022-08-02 10:55:30,942 MainThread INFO     --------------------------------------------------------------------------------
2022-08-02 10:55:30,942 MainThread INFO     Running Long System Test 0.1 - https://github.com/yugabyte/yb-long-system-test/
2022-08-02 10:55:30,942 MainThread INFO     --------------------------------------------------------------------------------
2022-08-02 10:55:30,942 MainThread INFO
2022-08-02 10:55:30,950 MainThread INFO     Reproduce with: git checkout 2027af8b && ./long_system_test.py --nodes=127.0.0.1:5433,127.0.0.2:5433,127.0.0.3:5433 --threads=10 --runtime=60 --max-columns=20 --complexity=full --seed=370648
2022-08-02 10:55:31,295 MainThread INFO     Database version: PostgreSQL 11.2-YB-2.15.2.0-b0 on x86_64-pc-linux-gnu, compiled by clang version 13.0.1 (https://github.com/yugabyte/llvm-project.git 191e3a05a55c8671bcc88d7387c04c55a4310500), 64-bit
2022-08-02 10:55:31,297 MainThread INFO     Creating tables for database db_lst_370648
Traceback (most recent call last):
  File "/home/deen/code/yb-long-system-test/./long_system_test.py", line 320, in <module>
    main()
  File "/home/deen/code/yb-long-system-test/./long_system_test.py", line 311, in main
    lst.run()
  File "/home/deen/code/yb-long-system-test/./long_system_test.py", line 233, in run
    self._create_tables()
  File "/home/deen/code/yb-long-system-test/./long_system_test.py", line 193, in _create_tables
    self.database.create(cur)
  File "/home/deen/code/yb-long-system-test/lst/dbobject.py", line 212, in create
    self.drop(cur)
  File "/home/deen/code/yb-long-system-test/lst/dbobject.py", line 219, in drop
    self.execute(cur, f'DROP DATABASE IF EXISTS {self.name()};')
  File "/home/deen/code/yb-long-system-test/lst/dbobject.py", line 17, in execute
    cur.execute(query)
psycopg2.OperationalError: server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.

2022-08-02 11:00:32,979 MainThread INFO
2022-08-02 11:00:32,979 MainThread INFO     --------------------------------------------------------------------------------
2022-08-02 11:00:32,979 MainThread INFO     Running Long System Test 0.1 - https://github.com/yugabyte/yb-long-system-test/
2022-08-02 11:00:32,979 MainThread INFO     --------------------------------------------------------------------------------
2022-08-02 11:00:32,979 MainThread INFO
2022-08-02 11:00:32,988 MainThread INFO     Reproduce with: git checkout 2027af8b && ./long_system_test.py --nodes=127.0.0.1:5433,127.0.0.2:5433,127.0.0.3:5433 --threads=10 --runtime=60 --max-columns=20 --complexity=full --seed=370648
2022-08-02 11:00:33,025 MainThread INFO     Database version: PostgreSQL 11.2-YB-2.15.2.0-b0 on x86_64-pc-linux-gnu, compiled by clang version 13.0.1 (https://github.com/yugabyte/llvm-project.git 191e3a05a55c8671bcc88d7387c04c55a4310500), 64-bit
2022-08-02 11:00:33,027 MainThread INFO     Creating tables for database db_lst_370648
Traceback (most recent call last):
  File "/home/deen/code/yb-long-system-test/./long_system_test.py", line 320, in <module>
    main()
  File "/home/deen/code/yb-long-system-test/./long_system_test.py", line 311, in main
    lst.run()
  File "/home/deen/code/yb-long-system-test/./long_system_test.py", line 233, in run
    self._create_tables()
  File "/home/deen/code/yb-long-system-test/./long_system_test.py", line 193, in _create_tables
    self.database.create(cur)
  File "/home/deen/code/yb-long-system-test/lst/dbobject.py", line 212, in create
    self.drop(cur)
  File "/home/deen/code/yb-long-system-test/lst/dbobject.py", line 219, in drop
    self.execute(cur, f'DROP DATABASE IF EXISTS {self.name()};')
  File "/home/deen/code/yb-long-system-test/lst/dbobject.py", line 17, in execute
    cur.execute(query)
psycopg2.errors.ObjectInUse: database "db_lst_370648" is being accessed by other users
DETAIL:  There are 2 other sessions using the database.
```
I will attach yugabyte-data directory. Available from within Yugabyte organization: https://drive.google.com/file/d/1FbAnHz317ChYeaICD8xd7WCWVpj3h3tj/view?usp=sharing
Might also be related to https://github.com/yugabyte/yugabyte-db/issues/13085, but now it's causing a FATAL.

**Comment by def-** [comment]
Still seeing this btw, assigning again, not sure why it was unassigned:
```
[deen@devp ~]$ cat /home/deen/yugabyte-data/node-1/disk-1/yb-data/tserver/logs/yb-tserver.FATAL.details.2022-12-23T23_48_25.pid3585381.txt /home/deen/yugabyte-data/node-1/disk-1/yb-data/tserver/logs/yb-tserver.FATAL.details.2022-12-24T02_29_58.pid3726978.txt
F20221223 23:48:25 ../../../../../../src/yb/yql/pggate/pggate.cc:1760] Check failed: _s.ok() Bad status: Invalid argument (yb/tserver/pg_client_service.cc:429): Unknown session: 851
    @     0x7fc24ac01e6a  google::LogMessage::SendToLog()
    @     0x7fc24ac028d0  google::LogMessage::Flush()
    @     0x7fc24ac051a9  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fc24f0623ba  yb::pggate::PgApiImpl::ClearSeparateDdlTxnMode()
    @     0x561708e3784b  YBResetDdlState
    @     0x561708e377b4  YBTxnDdlProcessUtility
    @     0x561708cc415a  PortalRunUtility
    @     0x561708cc3847  PortalRunMulti
    @     0x561708cc30f6  PortalRun
    @     0x561708cc0816  yb_exec_simple_query_impl
    @     0x561708cc0db8  yb_exec_query_wrapper_one_attempt
    @     0x561708cbde08  PostgresMain
    @     0x561708c2cbfc  BackendRun
    @     0x561708c2c043  ServerLoop
    @     0x561708c282db  PostmasterMain
    @     0x561708b85e9d  PostgresServerProcessMain
    @     0x561708b86622  main
    @     0x7fc24e947825  __libc_start_main
    @     0x56170892f7b9  _start
```

**Comment by fizaaluthra** [comment]
Closing as part of #13128  -- addressed in commit 5a6d54a6995b3d45dbba257abde296e91fa63e54
